src/main/java/eu/darkbot/captcha/CaptchaSolver.java

```diff
@@ -20,106 +22,133 @@
-                                    res.set((String) webView.getEngine().executeScript("grecaptcha.getResponse()"));
-                                });
-                                try {
-                                    Thread.sleep(400);
-                                } catch (InterruptedException e1) {
-                                    e1.printStackTrace();
-                                }
-                                if (res.get() != null && !res.get().isEmpty()) {
-                                    System.out.println("captcha key: " + res);
-                                    key = res.get();
-                                    Platform.exit();
-                                    for (Window w : JDialog.getWindows()) {
-                                        if (w instanceof JDialog)
-                                            if (((JDialog) w).getTitle().contains("Manual"))
-                                                w.setVisible(false);
-                                    }
-                                }
-                            }
-                        }).start();
-                    });
-                    final Timeline timeline = new Timeline(kf);
-                    Platform.runLater(timeline::play);
-                }
-            });
+            WebEngine engine = webView.getEngine();
 
-            webView.getEngine().setUserAgent("BigpointClient/1.5.2");
-            webView.getEngine().loadContent("<html><body></body></html>", "text/html; charset=utf8"); //avoid bug of javafx of user agent
-            webView.getEngine().load(url);
+            engine.getLoadWorker().stateProperty().addListener((obs, oldState, newState) -> {
+                if (newState != Worker.State.SUCCEEDED || !webView.getEngine().getLocation().equals(url)) return;
 
-        });
-        JPanel panel = new JPanel(new MigLayout("fill, insets 0"));
+                String script = "" +
+                        "function setup() {" +
+                        "let qcContainer = document.getElementById('qc-cmp2-container');" +
+                        "let container = document.getElementsByClassName('eh_mc_container eh_mc_table')[0];" +
+                        "var cap = document.getElementsByClassName('bgcdw_captcha')[0];" +
+                        "if (!qcContainer || !container || !cap) {" +
+                        "    setTimeout() => setup(), 5);" +
+                        "    return;" +
+                        "}" +
+                        "qcContainer.remove();" +
+                        "container.innerHTML = '';" +
+                        "container.appendChild(cap);" +
+                        "document.getElementsByTagName('body')[0].style.minWidth = 0;" +
+                        "let styles = [...document.getElementsByTagName('style')];" +
+                        "for (let i in styles) styles[i].remove();" +
+                        "}" +
+                        "setup();";
```

Comment:
new js is wrong and i think you cannot inject timer or async from jfx

Reply:
I think i fixed it now, it was missing a parenthesis